pkg/command/context.go

```diff
@@ -925,16 +925,34 @@ func deleteCtx(_ *cobra.Command, args []string) error {
 			return nil
 		}
 	}
+	ctx, err := config.GetContext(name)
+	if err != nil {
+		return err
+	}
 	installed, _, _, _ := getInstalledAndMissingContextPlugins() //nolint:dogsled
 	log.Infof("Deleting entry for context '%s'", name)
-	err := config.RemoveContext(name)
+	err = config.RemoveContext(name)
 	if err != nil {
 		return err
 	}
 	listDeactivatedPlugins(installed, name)
+	deleteKubeconfigContext(ctx)
+
 	return nil
 }
 
+func deleteKubeconfigContext(ctx *configtypes.Context) {
+	// Note: currently cleaning up the kubeconfig for TAE context types only.
+	// (Since the kubernetes context type can have kube context provided by the user, it may not be
+	// desired outcome for user if CLI deletes/cleanup kubeconfig provided by the user.)
+	if ctx.Target == configtypes.TargetTAE {
```

Comment:
Not blocking:

Now that https://github.com/vmware-tanzu/tanzu-cli/pull/535 also creates kube ctx synthesized from cluster endpoint and pinniped info in the default kubeconfig, it seems odd we are cleaning up TAE-based context on delete but not that. It might be good to add a comment/TODO to mention the possibility of addressing this in the future.

Reply:
Sure. I will add TODO comment and file an issue to address this in future